src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28060,6 +28060,67 @@ static void M(C c, string? p1, string? p2)
             CompileAndVerify(comp2);
         }
 
+        [Fact, WorkItem(49077, "https://github.com/dotnet/roslyn/issues/49077")]
+        public void NotNullIfNotNull_Return_Async()
```

Comment:
Consider adding a test like the following to demonstrate the analysis of parameters

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;
using System.Threading.Tasks;

public class C
{
    public async Task M([NotNullIfNotNull("s2")] string? s1, string? s2)
    {
        if (s2 is object)
        {
            return; // warning CS8824: Parameter 's1' must have a non-null value when exiting because parameter 's2' is non-null.
        }
    }
}
```
```